modules/network/src/main/java/org/apache/ignite/network/MessageSerializationRegistryImpl.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.network;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.ignite.network.serialization.MessageDeserializer;
+import org.apache.ignite.network.serialization.MessageSerializationFactory;
+import org.apache.ignite.network.serialization.MessageSerializationRegistry;
+import org.apache.ignite.network.serialization.MessageSerializer;
+
+/**
+ * Default implementation of a {@link MessageSerializationRegistry}.
+ */
+public class MessageSerializationRegistryImpl implements MessageSerializationRegistry {
+    /** message type -> MessageSerializerProvider instance */
+    private final Map<Integer, MessageSerializationFactory<?>> factories = new HashMap<>();
```

Comment:
I would still prefer a more effective data structure. The less maps we have, the better.

Reply:
okay...